gordon/interfaces.py

```diff
@@ -120,3 +120,41 @@ class IPublisherClient(IGenericPlugin):
         Args:
             event_msg (IEventMessage): Message ready to be sent to destination.
         """
+
+
+class IMetricRelay(Interface):
+    """Create and publish (push) metrics on demand."""
+
+    def incr(metric_name, value=1):
+        """Increment a metric. Create one if new."""
+
+    def timer(metric_name):
+        """Return a :class:`ITimerMetric`. Create one if new."""
+
+    def set_counter(metric_name, counter):
+        """Set a :class:`ICounterMetric` object to a given metric name."""
+
+    def set_timer(metric_name, timer):
+        """Set a :class:`ITimerMetric` to a given metric name."""
+
+
+class IGenericMetric(Interface):
+    """**Do not** implement this interface directly.
+
+    Use :class:`ICounterMetric` or :class:`ITimerMetric`
+    instead.
+    """
+
+
+class ICounterMetric(IGenericMetric):
+    """Monotonically increasing counter object."""
+
+    value = Attribute('Value, must start at 0.')
```

Comment:
:thinking: What do you think about changing this attribute from `value` to `counter`? Or adding that word to the attribute's description, eg. _Counter. Must start at 0._?

`value` is nice in that all the metric objects (only Counter and Timer now) have a uniform interface. But `counter` and `timing` would be more descriptive at the cost of less uniformity. Thoughts?

Reply:
I'm cool for updating the description, but changing the actual attribute name to be more specific per metric would make it a bit more difficult to work with. It allows us to not know which object type we're working with but still call something like `update`, much like how shumway works (i.e. [counter](https://github.com/spotify/shumway/blob/master/shumway/__init__.py#L68), [timer](https://github.com/spotify/shumway/blob/master/shumway/__init__.py#L84)).